File: frontend/src/api/summary.ts
export type GroupBy = "category" | "payment_method";

export interface SummaryGroup {
  id: string;
  name: string;
  color: string;
  amount: number;
  percentage: number;
}

export interface SummaryResponse {
  month: number;
  year: number;
  group_by: GroupBy;
  total: number;
  groups: SummaryGroup[];
}

const API_BASE = import.meta.env.VITE_API_BASE_URL ?? "/api";

export async function getExpenseSummary(
  month: number,
  year: number,
  groupBy: GroupBy
): Promise<SummaryResponse> {
  const params = new URLSearchParams({
    month: String(month),
    year: String(year),
    group_by: groupBy,
  });
  const resp = await fetch(`${API_BASE}/transactions/summary?${params}`, {
    headers: {
      "X-API-Key": import.meta.env.VITE_API_KEY ?? "",
    },
  });
  if (!resp.ok) {
    throw new Error(`Failed to fetch summary: ${resp.status}`);
  }
  return resp.json() as Promise<SummaryResponse>;
}

File: backend/app/schemas/summary.py
import uuid
from typing import Literal

from pydantic import BaseModel, Field


class SummaryQueryParams(BaseModel):
    month: int = Field(..., ge=1, le=12)
    year: int = Field(..., ge=1000, le=9999)
    group_by: Literal["category", "payment_method"]


class SummaryGroup(BaseModel):
    id: uuid.UUID
    name: str
    color: str
    amount: float
    percentage: float


class SummaryResponse(BaseModel):
    month: int
    year: int
    group_by: str
    total: float
    groups: list[SummaryGroup]
